### PR TITLE
Sonar and wall driver

### PR DESCRIPTION
**Pull request by JohnGilb**

### CODE REVIEW

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java [hunk 1]
```diff
@@ -50,13 +52,15 @@ public PoseSubsystem(WPIFactory factory, XPropertyManager propManager) {
         
         currentPitch = propManager.createEphemeralProperty("CurrentPitch", 0.0);
         currentRoll = propManager.createEphemeralProperty("CurrentRoll", 0.0);
+        frontDistance = propManager.createEphemeralProperty("FrontDistance", 0.0);
+        leftDistanceToWall = propManager.createEphemeralProperty("LeftDistanceToWall", 0.0);
     }
     
     public static class TemporaryVoltageMap
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: Do we know enough about the sensor we're using to do this more officially now?

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java [hunk 2]
```diff
@@ -90,10 +94,15 @@ public void setCurrentHeading(double headingInDegrees){
         currentHeading.setValue(headingInDegrees);
     }
     
+    public void updateRangefinders() {
+        frontDistance.set(frontDistanceSensor.getDistance());
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: would be better to call getFrontRangefinderDistance here instead?

File: CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToWallCommand.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        
+        this.requires(drive);
+    }
+    
+    @Override
+    public void initialize() {
+        wallManager.reset();
+    }
+    
+    public void setDesiredDistance(double distance) {
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToWallCommand.java]: distanceFromWall?

Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToWallCommand.java]: we've also been using target more than desired

File: CompetitionTemplate/src/competition/subsystems/drive/commands/UpdatePoseCommand.java [hunk 4]
```diff
@@ -30,6 +30,7 @@ public void initialize() {
     public void execute() {
         pose.updateCurrentHeading();
         defenseTraversal.measureState(Math.abs(pose.getRobotPitch()));
+        pose.updateRangefinders();
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/commands/UpdatePoseCommand.java]: Might be cleaner to just have a general `updateSensorState` on the pose subsystem and call that from here?

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java (hunk 1 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: Added other 3 sensors

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java (hunk 2 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: Good idea!

File: CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToWallCommand.java (hunk 3 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToWallCommand.java]: targetDistanceFromWall

File: CompetitionTemplate/src/competition/subsystems/drive/commands/UpdatePoseCommand.java (hunk 4 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/drive/commands/UpdatePoseCommand.java]: made a general updateAllSensors, which calls these helper methods (now made private)

**Pull request merged**